### PR TITLE
wails: revamp form password

### PR DESCRIPTION
**Pull request by maxime-carabina**
We are revamp the form password logic.

### CODE REVIEW

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 1]
```diff
@@ -52,78 +53,84 @@ function PasswordPrompt() {
...
+
+  function validate(e: SyntheticEvent) {
+    const form = parseForm(e);
+    const { password, passwordConfirm } = form;
+
+    if (!hasMoreThanFiveChars(password.toString())) {
+      setError({ password: 'Password must have at least 5 characters' });
+      return false;
+    } else if (
+      !hasSamePassword(password.toString(), passwordConfirm.toString())
```
Comment (mazmassa) [inline wails-frontend/src/pages/passwordPrompt.tsx:66]: Do we really need this casting to `.toString() !? They are already, no!? 🤔

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 2]
```diff
@@ -52,78 +53,84 @@ function PasswordPrompt() {
...
+        >
+          <h1 className="mas-title">{req.Msg}</h1>
+          <p className="mas-body pt-4">{getSubtitle()}</p>
           <div className="pt-4">
-            <Password
-              onChange={(e) => setPasswordConf(e.target.value)}
-              value={passwordConfirm}
-              name="input"
-              placeholder="Confirm your password"
+            <Input
```
Comment (mazmassa) [inline wails-frontend/src/pages/passwordPrompt.tsx:106]: I guess you can use `<Password>` here. The API for the error is already merged!

Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:106]: Yes sorry! I will correct this now.

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 1 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx:66]: Sorry it's mt fault. I will remove them.

**Review by peterjah — APPROVED** [review]
good job! just remove the .toString() and LFG!

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 3]
```diff
@@ -8,25 +8,27 @@ import {
 import { EventsOnce } from '../../wailsjs/runtime';
 import { events, promptAction, promptRequest } from '../events/events';
 import { handleApplyResult, handleCancel } from '../utils/utils';
+import { parseForm } from '../utils/parseForm';
+import { hasMoreThanFiveChars, hasSamePassword } from '../validation/password';
 
 import { FiLock } from 'react-icons/fi';
 import { Password, Button } from '@massalabs/react-ui-kit';
 
+interface ErrorObject {
```
Comment (mazmassa) [inline wails-frontend/src/pages/passwordPrompt.tsx:17]:
```suggestion
interface IErrorObject {
```

**Review by mazmassa — APPROVED** [review]
Left two comments... but overall LGTM!

Thanks for that Max! 🎉

File: wails-frontend/src/pages/passwordPrompt.tsx (hunk 3 above)
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx]: Done

File: wails-frontend/src/pages/passwordPrompt.tsx [hunk 4]
```diff
@@ -52,78 +54,82 @@ function PasswordPrompt() {
...
+      : ApplyPassword(password);
+  }
+
+  async function handleSubmit(e: SyntheticEvent) {
+    e.preventDefault();
+
+    if (validate(e)) {
+      save(e);
     }
   }
```
[the remark this answers is not in the document]
Comment (maxime-carabina) [reply wails-frontend/src/pages/passwordPrompt.tsx]: Done

**Pull request merged**